internal/git/git.go

```diff
@@ -0,0 +1,130 @@
+package git
+
+import (
+	"emperror.dev/errors"
+	"github.com/sirupsen/logrus"
+	"io"
+	"os/exec"
+	"strings"
+)
+
+// Missing is a sentinel zero-value for object id (aka sha).
+// Git treats this value as "this thing doesn't exist".
+// For example, when updating a ref, if the old value is specified as EmptyOid,
+// Git will refuse to update the ref if already exists.
+const Missing = "0000000000000000000000000000000000000000"
+
+type Repo struct {
+	repoDir       string
+	defaultBranch string
+	log           logrus.FieldLogger
+}
+
+func OpenRepo(repoDir string) (*Repo, error) {
+	r := &Repo{
+		repoDir, "",
+		logrus.WithFields(logrus.Fields{"repo": repoDir}),
+	}
+
+	for _, possibleTrunk := range []string{"main", "master", "default", "trunk"} {
+		_, err := r.Git("rev-parse", "refs/remotes/origin/"+possibleTrunk)
+		if err == nil {
+			r.defaultBranch = possibleTrunk
+			break
+		}
+	}
+	return r, nil
+}
+
+func (r *Repo) DefaultBranch() string {
+	return r.defaultBranch
+}
+
+func (r *Repo) Git(args ...string) (string, error) {
+	cmd := exec.Command("git", args...)
+	cmd.Dir = r.repoDir
```

Comment:
why do you need to specify a directory? Assuming someone is running this from the git repo, does it matter where in repo they are?

Reply:
See comment on `-C` above.